pyproject.toml

```diff
@@ -2,11 +2,7 @@
 # Copyright © 2024 Frequenz Energy-as-a-Service GmbH
 
 [build-system]
-requires = [
-  "setuptools == 68.1.0",
-  "setuptools_scm[toml] == 7.1.0",
-  "frequenz-repo-config[lib] == 0.9.1",
```

Comment:
Is not removed, it is changed to remove `repo-config` (look at the added line below). The format is also changed to be a one-liner now, I have automatic formatting using [taplo](https://taplo.tamasfe.dev/) (which I want to eventually [integrate to repo-config](https://github.com/frequenz-floss/frequenz-repo-config-python/issues/15)).

Including `repo-config` as a build dependency in every project was actually a mistake, it was only needed for API projects and now that this is taken care by setuptools-betterproto (if we do the switch), it will not be used by any project. My idea is to split repo-config even further in the future, it was an experiment and now that we have a much clear picture about what we need from it, it is a good time to split it into smaller projects and use those smaller projects only where necessary.

Reply:
It is unrelated to this PR, though, I just did it now because I realized it was a mistake to include `repo-config` in the build dependencies right after creating `setuptools-betterproto`.